sql/planbuilder/from.go

```diff
@@ -658,6 +658,12 @@ func (b *Builder) buildResolvedTable(inScope *scope, db, schema, name string, as
 		if !schemaFound {
 			b.handleErr(sql.ErrDatabaseSchemaNotFound.New(schema))
 		}
+	} else if isScd && schema == "" && b.currentDatabase != nil {
```

Comment:
I would split this condition up. In the postgres context, there's always a current database, it's a programming error for there to not be one.